### PR TITLE
Feature/jrh models

### PR DESCRIPTION
**Pull request by jrhorn424**
Please see commit messages for detail. Comments welcome.

### CODE REVIEW

File: app/models/field_experiment.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class FieldExperiment < Experiment
```
Comment (mlfreer) [inline app/models/field_experiment.rb:1]: Are you sure that we need separate tables for each kind of experiment and can not just to have an additional row in the experiments table?
Cause in the future we can have an necessity to add other types of experiments (as we have to schedule test-fests now)

Comment (jrhorn424) [inline app/models/field_experiment.rb:1]: Good question. That isn't another table, just another object. The type is stored in the experiments table. In this case, `FieldExperiment.new` would create an `Experiment` with `Experiment.type # => "FieldExperiment"`.

**Comment by jrhorn424** [comment]
Since STI is used for two features, please note in particular the commit message for 9a396e90a9c3fb215a6588e84193e9e4248b7ce3.

File: app/models/field_experiment.rb (hunk 1 above)
Comment (mlfreer) [inline app/models/field_experiment.rb:1]: great!
but if we will have to add some labs or experiments we'll have to get into the code?
or you'll be able to create the page which will create the class? (it is a reason why i was for the attribute) and the same for the labs?

Comment (jrhorn424) [inline app/models/field_experiment.rb:1]: On 11 Mar 2014, at 4:00, mlfreer wrote:

> > @@ -0,0 +1,2 @@
> > +class FieldExperiment < Experiment
> 
> great!
> but if we will have to add some labs or experiments we'll have to get 
> into the code?
> or you'll be able to create the page which will create the class? (it 
> is a reason why i was for the attribute) and the same for the labs?

One of the things I considered when making this design choice was the 
likelihood and cost of change. Labs are setup as a traditional table and 
are easy to change via traditional CRUD actions through a webpage. 
Experiment types aren't something that change. Do we have another 
methodology for experiments besides field experiments, lab experiments, 
and surveys? If we need one in the future it's trivial to add the empty 
class in a new file and deploy, and STI is not something that's likely 
to break. What it does buy us is the ability to define new behavior on 
the subclasses while sharing the data between them.

## 

Jeffrey Horn
http://hello.jrhorn.me/

**Pull request merged**